ddtrace/contrib/asyncpg/patch.py

```diff
@@ -0,0 +1,199 @@
+        db.NAME: params.database,
+    }
+
+
+class _TracedConnection(wrapt.ObjectProxy):
+    def __init__(self, conn, pin):
+        super(_TracedConnection, self).__init__(conn)
+        pin.onto(self)
+
+    async def _traced_method(self, method, query, args, kwargs):
+        pin = Pin.get_from(self)
+        if not pin or not pin.enabled:
+            log.debug("No pin or pin disabled, skipping tracing for %s", method)
+            return await method(*args, **kwargs)
+
+        with pin.tracer.trace(
+            "postgres.query", resource=query, service=ext_service(pin, config.asyncpg), span_type=SpanTypes.SQL
+        ) as span:
+            span.set_tag(SPAN_MEASURED_KEY)
+            span.set_tags(_get_connection_tags(self))
+            return await method(*args, **kwargs)
+
+    async def execute(self, *args, **kwargs):
+        query = get_argument_value(args, kwargs, 0, "query")
+        return await self._traced_method(self.__wrapped__.execute, query, args, kwargs)
+
+    async def executemany(self, *args, **kwargs):
+        command = get_argument_value(args, kwargs, 0, "command")
+        return await self._traced_method(self.__wrapped__.executemany, command, args, kwargs)
+
+    async def fetch(self, *args, **kwargs):
+        query = get_argument_value(args, kwargs, 0, "query")
+        return await self._traced_method(self.__wrapped__.fetch, query, args, kwargs)
+
+    async def fetchval(self, *args, **kwargs):
+        query = get_argument_value(args, kwargs, 0, "query")
+        return await self._traced_method(self.__wrapped__.fetchval, query, args, kwargs)
+
+    async def fetchrow(self, *args, **kwargs):
+        query = get_argument_value(args, kwargs, 0, "query")
+        return await self._traced_method(self.__wrapped__.fetchrow, query, args, kwargs)
+
+
+@with_traced_module
+async def _traced_connect(asyncpg, pin, func, instance, args, kwargs):
+    with pin.tracer.trace(
+        "postgres.connect", span_type=SpanTypes.SQL, service=ext_service(pin, config.asyncpg)
+    ) as span:
+        conn = _TracedConnection(await func(*args, **kwargs), pin)
+        span.set_tags(_get_connection_tags(conn))
+        return conn
+
+
+class _TracedCursorFactory(wrapt.ObjectProxy):
+    def __init__(self, cursor, pin):
+        super(_TracedCursorFactory, self).__init__(cursor)
+        pin.onto(self)
+
+    def __aiter__(self):
+        return self.__wrapped__.__aiter__()
+
+    def __await__(self):
+        pin = Pin.get_from(self)
+        if not pin or not pin.enabled:
+            log.debug("No pin or pin disabled, skipping tracing")
+            return self.__wrapped__.__await__()
+
+        with pin.tracer.trace(
+            "postgres.cursor", resource=self._query, span_type=SpanTypes.SQL, service=ext_service(pin, config.asyncpg)
+        ):
+            return self.__wrapped__.__await__()
+
+
+@with_traced_module_sync
+def _traced_connection_cursor(asyncpg, pin, func, instance, args, kwargs):
```

Comment:
some docstrings would be nice, at least what is the target for these.

I am finding it hard to scan the doc with "traced connection" "traced cursor" "traced connect" "traced c..."